### DISCUSSION TITLE
Fixing EPUB files before send.

### DISCUSSION THREAD

**Discussion by NeoDuder** [crocodilestick/Calibre-Web-Automated#147] [Q&A]
Hey folks,

Just installed this and loving it so far. Recently I've been getting issues with downloaded EPUB files not sending to Kindle with an "E999 – Send to Kindle Internal Error". Investigating this issue is how I found CWA in the first place. I originally solved it by manually converting the EPUB to MOBI and then back again, but even this left a few files unable to send due to a missing language field in the metadata, requiring another manual edit.

Is there a way to get CWA to check that the metadata is in a state that 'Send to Kindle' will work without all the manual steps?

Thanks in advance :)

Chris

**Comment by crocodilestick** [comment]
Hi @NeoDuder ! There is though it is coming in the next version. The script that powers this site [https://kindle-epub-fix.netlify.app/](https://kindle-epub-fix.netlify.app/) has been converted to python and is being integrated into the most recent dev builds.

From the new release, CWA will automatically use this script to check any processed epub for ecoding errors ect that would cause Send-to-Kindle to reject the file and correct any issues automatically

**Reply by NeoDuder** [reply]
Excellent, thanks for the update :)
